control/controlknobs/controlknobs.go

```diff
@@ -81,6 +81,15 @@ type Knobs struct {
 	// how to dial the destination address. When true, it also makes the DNS forwarder
 	// use UserDial instead of SystemDial when dialing resolvers.
 	UserDialUseRoutes atomic.Bool
+
+	// DisableSplitDNSWhenNoCustomResolvers indicates that the node's DNS manager
+	// should not adopt a split DNS configuration even though the Config of the
+	// resolver only contains routes that do not specify custom resolver(s), hence
+	// all DNS queries can be safely sent to the upstream DNS resolver and the
+	// node's DNS forwarder doesn't need to handle all DNS traffic.
+	// This is for now (2024-06-06) an iOS-specific battery life optimization,
+	// and this knob allows us to disable the optimization remotely if needed.
```

Comment:
I'm still having trouble reading this.

Is the plan to change the behavior on iOS by default and rely on this as an safety rollback mechanism to revert back to current behavior, or does this knob enable the new behavior?

There are so many negated words in here (disable, when no, should not, allows us to disable ...) that I'm lost.

The plan's for MagicDNS suffixes (*.ts.net) to go to 100.100.100.100 and everything else to be handled by the system, right? Does this turn on that behavior, or off?